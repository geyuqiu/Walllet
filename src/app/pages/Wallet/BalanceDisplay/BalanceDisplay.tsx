import {BigNumber} from '@arkecosystem/platform-sdk-support';
import {Icon} from "app/components/Icon/Icon";
import React from "react";

import {Amount} from '../../../components/Amount/Amount';
import {Circle} from '../../../components/Circle/Circle';
import {Wallet} from '../model';

type BalanceDisplayProps = {
	wallet: Wallet | null;
}

export const BalanceDisplay = ({wallet}: BalanceDisplayProps) => (
		<div className="flex flex-row mt-6 lg:mt-0">
			<div className="hidden sm:block">
				<Circle className="border rounded-full mr-3" size="lg">
						<Icon name="Balance" width={15} height={15} fill='#000000' stroke='#FBC457'/>
				</Circle>
			</div>
			<div className="flex flex-col">
				<span className="text-gray-darkest">Balance</span>
				{wallet?.balance && <Amount
					data-testid="balance__amount"
					value={BigNumber.make(wallet.balance)}
					className="font-bold"
				/>}
			</div>
		</div>
	);
