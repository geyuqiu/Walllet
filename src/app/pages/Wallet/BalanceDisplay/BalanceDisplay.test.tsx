import {render} from "@testing-library/react";
import React from "react";

import {BalanceDisplay} from "./BalanceDisplay";

describe("BalanceDisplay", () => {

	it("valid wallet should match snapshot", () => {
		const { container } = render(<BalanceDisplay wallet={{
			address: 'AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW',
			balance: '0'
		}}/>);

		expect(container).toMatchSnapshot();
	});

	it("wallet is null, should match snapshot", () => {
		const { container } = render(<BalanceDisplay wallet={null}/>);

		expect(container).toMatchSnapshot();
	});
});
