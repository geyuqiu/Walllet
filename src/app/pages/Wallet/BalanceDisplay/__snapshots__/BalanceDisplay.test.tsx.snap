// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceDisplay valid wallet should match snapshot 1`] = `
<div>
  <div
    class="flex flex-row mt-6 lg:mt-0"
  >
    <div
      class="hidden sm:block"
    >
      <div
        class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy border rounded-full mr-3 ring-theme-background"
      >
        <div
          class="sc-bdfBwQ dfWOLJ"
          height="15"
          width="15"
        >
          <svg
            fill="#000000"
            stroke="#FBC457"
          >
            balance.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <span
        class="text-gray-darkest"
      >
        Balance
      </span>
      <span
        class="font-bold"
        data-testid="balance__amount"
      >
        0 DARK
      </span>
    </div>
  </div>
</div>
`;

exports[`BalanceDisplay wallet is null, should match snapshot 1`] = `
<div>
  <div
    class="flex flex-row mt-6 lg:mt-0"
  >
    <div
      class="hidden sm:block"
    >
      <div
        class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy border rounded-full mr-3 ring-theme-background"
      >
        <div
          class="sc-bdfBwQ dfWOLJ"
          height="15"
          width="15"
        >
          <svg
            fill="#000000"
            stroke="#FBC457"
          >
            balance.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <span
        class="text-gray-darkest"
      >
        Balance
      </span>
    </div>
  </div>
</div>
`;
